### PR TITLE
[ETCM-205] Add information about checkpoints in tx history

### PR DESCRIPTION
**Pull request by kapke**
# Description

This PR adds information about tx being in checkpointed block or not to `mantis_getAccountTransactions`.
Also, as it is very useful - in this PR I've added a dependency on diffx - library which highlights the differences in compared data.

# Testing

All tests should pass. One can also try to use `qa_generateCheckpoint` for adding a checkpoint and checking if transactions from lower blocks are properly marked

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -54,7 +54,9 @@ object Dependencies {
     "org.scalatest" %% "scalatest" % "3.2.2" % "it,test",
     "org.scalamock" %% "scalamock" % "5.0.0" % "test",
     "org.scalatestplus" %% "scalacheck-1-14" % "3.2.2.0" % "test",
-    "org.scalacheck" %% "scalacheck" % "1.14.3" % "it,test"
+    "org.scalacheck" %% "scalacheck" % "1.14.3" % "it,test",
+    "com.softwaremill.diffx" %% "diffx-core" % "0.3.30" % "test",
+    "com.softwaremill.diffx" %% "diffx-scalatest" % "0.3.30" % "test"
```
Comment (mmrozek) [inline project/Dependencies.scala:59]: Don't we need to update nix dependencies?

File: src/main/scala/io/iohk/ethereum/domain/Address.scala [hunk 2]
```diff
@@ -33,6 +36,12 @@ object Address {
     require(bytes.length <= Length, s"Invalid address: $hexString")
     Address(bytes)
   }
+
+  def apply(keyPair: AsymmetricCipherKeyPair): Address = {
+    //byte 0 of encoded ECC point indicates that it is uncompressed point, it is part of bouncycastle encoding
+    val pub = keyPair.getPublic.asInstanceOf[ECPublicKeyParameters].getQ.getEncoded(false).tail
```
Comment (mmrozek) [inline src/main/scala/io/iohk/ethereum/domain/Address.scala:42]: Couldn't we use `crypto.pubKeyFromKeyPair`?

File: project/Dependencies.scala (hunk 1 above)
Comment (kapke) [reply project/Dependencies.scala:59]: We do, I wanted buildkite to generate proper updates for me

File: src/main/scala/io/iohk/ethereum/domain/Address.scala (hunk 2 above)
Comment (kapke) [reply src/main/scala/io/iohk/ethereum/domain/Address.scala:42]: Good point!

**Review by mmrozek — APPROVED** [review]
LGTM!

File: src/main/scala/io/iohk/ethereum/domain/Address.scala [hunk 3]
```diff
@@ -33,6 +35,11 @@ object Address {
     require(bytes.length <= Length, s"Invalid address: $hexString")
     Address(bytes)
   }
+
+  def apply(keyPair: AsymmetricCipherKeyPair): Address = {
+    val pub = crypto.pubKeyFromKeyPair(keyPair)
+    Address(crypto.kec256(pub).drop(FirstByteOfAddress))
+  }
```
Comment (lemastero) [inline src/main/scala/io/iohk/ethereum/domain/Address.scala:42]: 👍

File: src/main/scala/io/iohk/ethereum/domain/Address.scala [hunk 4]
```diff
@@ -33,6 +35,11 @@ object Address {
     require(bytes.length <= Length, s"Invalid address: $hexString")
     Address(bytes)
   }
+
+  def apply(keyPair: AsymmetricCipherKeyPair): Address = {
+    val pub = crypto.pubKeyFromKeyPair(keyPair)
+    Address(crypto.kec256(pub).drop(FirstByteOfAddress))
```
Comment (KonradStaniec) [inline src/main/scala/io/iohk/ethereum/domain/Address.scala:41]: minor:  `.drop(FirstByteOfAddress)` is probably not necessary , as `Address` bytesting constructor takes only last 20byte also `FirstByteOfAddress` is weird name for something which drops 12 bytes

**Review by KonradStaniec — APPROVED** [review]
Only 1 minor comment

**Review by lemastero — APPROVED** [review]

**Pull request merged**